### PR TITLE
Adding Github action workflow for PR builds

### PR DESCRIPTION
**Pull request by VibhavariMandal-sumo**
This is in continuation of https://github.com/open-telemetry/opamp-java/pull/5. The workflow triggers when a pull request is raised.

This checks out the proto submodules, builds the proto repo & includes all the proto classes in the final jar of opamp-java-library.

After building the library, it is published to a maven local repository. For publishing to a remote maven repo, I suggest we do it in a separate PR, once we finalise where we need to publish.

vmandal@vmandal-mac opamp-java % tree build/repos/snapshots/io/opentelemetry/opamp-java/1.0.0-SNAPSHOT/
build/repos/snapshots/io/opentelemetry/opamp-java/1.0.0-SNAPSHOT/
├── maven-metadata.xml
├── maven-metadata.xml.md5
├── maven-metadata.xml.sha1
├── maven-metadata.xml.sha256
├── maven-metadata.xml.sha512
├── opamp-java-1.0.0-20221212.134009-1-javadoc.jar
├── opamp-java-1.0.0-20221212.134009-1-javadoc.jar.md5
├── opamp-java-1.0.0-20221212.134009-1-javadoc.jar.sha1
├── opamp-java-1.0.0-20221212.134009-1-javadoc.jar.sha256
├── opamp-java-1.0.0-20221212.134009-1-javadoc.jar.sha512
├── opamp-java-1.0.0-20221212.134009-1-sources.jar
├── opamp-java-1.0.0-20221212.134009-1-sources.jar.md5
├── opamp-java-1.0.0-20221212.134009-1-sources.jar.sha1
├── opamp-java-1.0.0-20221212.134009-1-sources.jar.sha256
├── opamp-java-1.0.0-20221212.134009-1-sources.jar.sha512
├── opamp-java-1.0.0-20221212.134009-1.jar
├── opamp-java-1.0.0-20221212.134009-1.jar.md5
├── opamp-java-1.0.0-20221212.134009-1.jar.sha1
├── opamp-java-1.0.0-20221212.134009-1.jar.sha256
├── opamp-java-1.0.0-20221212.134009-1.jar.sha512
├── opamp-java-1.0.0-20221212.134009-1.module
├── opamp-java-1.0.0-20221212.134009-1.module.md5
├── opamp-java-1.0.0-20221212.134009-1.module.sha1
├── opamp-java-1.0.0-20221212.134009-1.module.sha256
├── opamp-java-1.0.0-20221212.134009-1.module.sha512
├── opamp-java-1.0.0-20221212.134009-1.pom
├── opamp-java-1.0.0-20221212.134009-1.pom.md5
├── opamp-java-1.0.0-20221212.134009-1.pom.sha1
├── opamp-java-1.0.0-20221212.134009-1.pom.sha256
└── opamp-java-1.0.0-20221212.134009-1.pom.sha512

### CODE REVIEW

**Comment by VibhavariMandal-sumo** [comment]
@jack-berg @jlegoff , this PR intends to build and publish the final jar to a local maven repo on raising a pull request. 

1. I have addressed most of the comments you provided  in https://github.com/open-telemetry/opamp-java/pull/5
2. This is the first part of  complete CI/CD setup.
3. This will unblock the opamp-java development.
4. I will raise 2 more PRs for main build workflow and release build workflow.
5. In future PRs , I will add publishing to a remote maven repo.

Request you to review.

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+        with:
+          arguments: jar
+
+      - name: Publish Jar with Gradle
+        uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+        with:
+          arguments: publish
+
+      - name: Find the path of built jar
```
Comment (jlegoff) [inline .github/workflows/pr-build.yml]: what is the purpose of this step and the following one? Why do we need to extract the jar and see the contents/

Comment (VibhavariMandal-sumo) [reply .github/workflows/pr-build.yml]: @jlegoff this was more for debugging whether all the compiled classes got included in the final jar or not.

Comment (VibhavariMandal-sumo) [reply .github/workflows/pr-build.yml]: But yes, I will remove these steps

Comment (VibhavariMandal-sumo) [reply .github/workflows/pr-build.yml]: Removed

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+          settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+      - name: Check if submodule has been pulled
```
Comment (jlegoff) [inline .github/workflows/pr-build.yml]: not sure if this is needed. Won't the build fail if the submodule isn't pulled?

**Review by jlegoff — APPROVED** [review]
Left a minor comment, otherwise it looks good to me

File: .github/workflows/pr-build.yml (hunk 2 above)
Comment (VibhavariMandal-sumo) [reply .github/workflows/pr-build.yml]: no, it wont fail, just that opamp-proto jar will not be included in the final jar

Comment (jlegoff) [reply .github/workflows/pr-build.yml]: in this case, I would remove this step

**Comment by VibhavariMandal-sumo** [comment]
@jlegoff removed the step to check if submodule has been pulled

**Review by tigrannajaryan — APPROVED** [review]
Thank you @VibhavariMandal-sumo and @jlegoff

**Pull request merged**